azurerm/data_source_image.go

```diff
@@ -134,28 +135,24 @@ func dataSourceArmImageRead(d *schema.ResourceData, meta interface{}) error {
 		r := regexp.MustCompile(nameRegex.(string))
 
 		list := []compute.Image{}
-		resp, err := client.ListByResourceGroup(resGroup)
+		resp, err := client.ListByResourceGroupComplete(ctx, resGroup)
 		if err != nil {
-			if utils.ResponseWasNotFound(resp.Response) {
+			if utils.ResponseWasNotFound(resp.Response().Response) {
 				d.SetId("")
 				return nil
 			}
 			return fmt.Errorf("[ERROR] Error getting list of images (resource group %q): %+v", resGroup, err)
 		}
-		for _, ri := range *resp.Value {
-			if r.Match(([]byte)(*ri.Name)) {
-				list = append(list, ri)
+
+		for resp.NotDone() {
```

Comment:
Do you know if the `resp` object will abort if the cancel on the `StopContext` is invoked? If not, we may want a `select` in here on the `StopContext.Done`, but not a big deal, I don't think it should hold up this pr as its an edge case to have enough images where it would be an issue I think.

Reply:
not from what I can see, whilst it's a valid concern it's such an edge case (and minor issue in terms of iterations) I think we can safely ignore it for the moment